### PR TITLE
fix(aws-cloudfront-s3) cfn nag error when using existingBucketInterface

### PR DESCRIPTION
**Pull request by hnishar**
*Issue #, if available:* [318](https://github.com/awslabs/aws-solutions-constructs/issues/318)

*Description of changes:*

Fix Cfn Nag error when passing  existingBucketInterface to aws-cloudfront-s3 construct

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 844a33423802517159c08d728d998fa1c370a9b4
* Result: FAILED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3011f844-888c-4309-9748-93e6c8e4d368%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/patterns/@aws-solutions-constructs/core/lib/cloudfront-distribution-helper.ts [hunk 1]
```diff
@@ -125,13 +125,14 @@ export function CloudFrontDistributionForS3(scope: cdk.Construct,
...
   // Extract the CfnBucketPolicy from the sourceBucket
   const bucketPolicy = sourceBucket.policy as s3.BucketPolicy;
-  addCfnSuppressRules(bucketPolicy, [
-    {
-      id: 'F16',
-      reason: `Public website bucket policy requires a wildcard principal`
-    }
-  ]);
-
+  if (bucketPolicy) {
```
Comment (biffgaut) [inline source/patterns/@aws-solutions-constructs/core/lib/cloudfront-distribution-helper.ts:128]: Let's add a comment explaining that the lack of a bucketPolicy means the bucket was imported from outside the stack so the lack of cfn_nag suppression is not an issue.

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: b32353dbc8af96157c583f43b11f6d9f2828fa1d
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5dae198d-f77a-47d4-b677-9a7d78a9cea7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 7fc4a2d517fb6936711855765bb1664430e32c33
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b23556cc-4224-4a27-9d5a-5417cf61614d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 567bcb07dc17769a4d7937578aeda402db9f3a06
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1f490bfd-3960-453c-985c-91619b61fdd6%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Pull request merged**